data/data_controller.cpp

```diff
@@ -3,34 +3,101 @@
 #include <ctime>
 #include <chrono>
 #include <memory>
+#include <iostream>
+#include <assert.h>
+#include <unordered_map>
 #include "elements/Timer.h"
 
+
 namespace data {
-    static std::chrono::time_point g_appStart = std::chrono::system_clock().now();
-    static auto clock_now = std::chrono::system_clock::now();
-    static auto system_now = std::chrono::system_clock::now();
-
-    DataController::DataController() {
-        using namespace std::chrono_literals;
-        // active timers
-        for (int i = 0; i < 3; i++) {
-            auto new_timer = std::make_shared<Timer>( data::Timer(id_count++, g_appStart, "Purple", "Quota completed and other business idk.", g_appStart) );
-            m_timers.insert(new_timer);
-        }
-        // completed timers
-        for (int i = 0; i < 15; i++) {
-            std::chrono::system_clock::time_point end{ g_appStart + 2111h + 15s };
-            auto new_timer = std::make_shared<Timer>( data::Timer(id_count++, g_appStart, "Studying", "Quota completed and other business idk.", g_appStart));
-            new_timer->stop();
-            m_timers.insert(new_timer);
-        }
+    DataController::DataController() : m_createTable{ nullptr }, m_timerInsert{ nullptr }, m_timerUpdate{ nullptr }, m_selectAll{ nullptr } {
+        int rc = sqlite3_open_v2("emiHD.db", &m_dbHandle, SQLITE_OPEN_READWRITE | SQLITE_OPEN_CREATE, nullptr);
+        sqlite3_prepare_v2(m_dbHandle, DB_SCHEMA.c_str(), DB_SCHEMA.size() + 1, &m_createTable, nullptr);
+        sqlite3_step(m_createTable);
     }
     DataController::~DataController() {
+        sqlite3_finalize(m_createTable);
+        sqlite3_finalize(m_timerInsert);
+        sqlite3_finalize(m_timerUpdate);
+        sqlite3_finalize(m_selectAll);
+        sqlite3_close(m_dbHandle);
     }
 
 
     void DataController::createTimer(std::string title) {
+        int rc = SQLITE_OK;
+        if (m_timerInsert == nullptr) {
+            rc = sqlite3_prepare_v2(m_dbHandle, INSERT_QUERY.c_str(), INSERT_QUERY.size() + 1, &m_timerInsert, nullptr);
+            assert (rc == SQLITE_OK);
+        }
+        sqlite3_reset(m_timerInsert);
         auto now = std::chrono:: system_clock::now();
-        m_timers.insert(std::make_shared<Timer>(data::Timer(id_count++, now, title, "", now)));
+        // Of format:
+        //    INSERT INTO TIMERS (TITLE, START_DATE, STATE, DURATION, ELAPSED, PAUSED) VALUES (?, ?, ?, ?, ?, ?)
```

Comment:
Look for an opportunity to allow a Timer to bind itself to the query. New barebones Timer constructor so that it may be passed into the instance? Or use a static method?